core/shared/src/test/scala/fs2/StreamCombinatorsSuite.scala

```diff
@@ -371,7 +372,9 @@ class StreamCombinatorsSuite extends Fs2Suite {
 
     test("filters up to N items in parallel") {
       val s = Stream.range(0, 100)
-      val n = 5
+      val n =
+        if (isJVM) 5
+        else 2 // for some reason JS is running evalFilterAsync with only 2 concurrent level
```

Comment:
Can't get why, but with the new `cats-effect` version `evalFilterAsync` and `evalFilterNotAsync` is running with max concurrent level 2 on JS (> 2 is leads to broken tests).

Reply:
https://github.com/typelevel/fs2/issues/2593

Reply:
I've tried to catch this in `main` initially - have no results. So have concluded that's because of a bump of the `cats-effect`. Thanks @vasilmkd

Reply:
Yeah, Cats Effect 3.3 will bring changes to the JS specific code, and it seems to negatively impact this issue. If your fix works, I think it's fine.

Reply:
Yep, it works (at least it's not failed in 5-10 runs).

Reply:
I was able to bisect this to my PR adding tracing for JS: https://github.com/typelevel/cats-effect/pull/2207. This seems to be another victim of the performance hit caused by tracing. Increasing the sleep fixes it for me locally, try applying this patch: https://github.com/typelevel/fs2/commit/843ff4a3ce65691a79e9dea8654b115057af643f.

Reply:
The patch https://github.com/typelevel/fs2/commit/843ff4a3ce65691a79e9dea8654b115057af643f was applied.